### DISCUSSION TITLE
Spellcheck is not working for some windows 10 users in CEF Minimal Example

### DISCUSSION THREAD

**Discussion by vivekbhat31** [cefsharp/CefSharp#3518] [General]
- **What version of the product are you using?**
    v88.2.90
- **What architecture x86 or x64?**
x64
	
- **What version of .Net?**
.Net 5.0
    
- **On what operating system?**
Win10

- **Are you using `WinForms`, `WPF` or `OffScreen`?**
WPF
    
- **What steps will reproduce the problem?**
   We are not able to repro this issue in all the machines. Only in some machines we see that Spellcheck is not working in CEF Minimal Example

- **What is the expected output? What do you see instead?**
- Spell check should work in all machines

- **Please provide any additional information below.**
    We've tested in simple WPF application and Spellcheck is working in it. In Chrome browser too it's working fine. Only in CEF Minimal Example we see this issue.

- **Does this problem also occur in the `CEF` Sample Application**
    We've tested on CEF Minimal Example and there we see this issue.

**Comment by amaitland** [comment]
> What version of the product are you using?
> v88.2.90

Does the problem reproduce with version 89?

Did you check the log file? 

Do the machines have internet access to download the dictionary? 

What's different about these machines to the ones that work? I've not seen any other reports of problems. 

There is an known upstream issue https://bitbucket.org/chromiumembedded/cef/issues/3055/windows-spell-checker-not-working-add

CefSharp has the workaround applied by default https://github.com/cefsharp/CefSharp/blob/cefsharp/89/CefSharp.Core.Runtime/CefSettingsBase.h#L59


> We've tested on CEF Minimal Example and there we see this issue.

Please clarify what you mean by CEF minimal example? Do you mean the CEF Sample application? What command line args did you test with? 

I'm converting this to a discussion as it's unlikely a bug in CefSharp directly (requiring a code change) as spell checking is entirely implemented in CEF/Chromium.

**Reply by vivekbhat31** [reply]
This most probably will happen on ver 89 as well, as it was working for us on ver 88.
Yes, we checked log file, but it doesn't have any logs for spellchecker.
Internet access is present as pages are loading fine, only issue with spellchecking on any page.
We didn't find any noticeable differences in 2 machines, all settings are same.

CEFSharp Minimal Example, where we see this issue. 

Can you please let us know, how spellcheck  works in Cef & what troubleshooting we can do to check if any components required for it is present or not on the system.

**Reply by amaitland** [reply]
> This most probably will happen on ver 89 as well, as it was working for us on ver 88.

Please test to confirm. Each version of chromium brings a different set of characteristics. Do this first. You can test with the minimal example. 

> We didn't find any noticeable differences in 2 machines, all settings are same.

As it works on most machines a difference in your environment in quite likely the case, I'd suggest doing a deeper comparison. Are they running the exact same windows build? 

> Can you please let us know, how spellcheck works in Cef & what troubleshooting we can do to check if any components required for it is present or not on the system.

Manually check to see if the dictionary files have been download, if not copy them from a working machine. I don't remember the exact location, you'll have to search on https://magpcss.org/ceforum/index.php to find the details.
